pkg/proxy/kafka.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/cilium/cilium/pkg/kafka"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/nodeaddress"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/proxy/accesslog"
+
+	"github.com/optiopay/kafka/proto"
+	log "github.com/sirupsen/logrus"
+)
+
+// kafkaRedirect implements the Redirect interface for an l7 proxy
+type kafkaRedirect struct {
+	lock.RWMutex
```

Comment:
Is this lock for the whole structure or the `rules    policy.L7DataMap`? Please write the lock before the attribute that is protecting.

Reply:
My guess is that it's for whole struct since it's embedded, but I agree that it would be nice to have some more verbosity here.

Reply:
That's my question. I don't know if it is because the lock seems to be only used in for the `rules` attribute and not for the remaining attributes.

Reply:
fixed